### PR TITLE
Feature/ET-1607 stripe customer name

### PR DESCRIPTION
**Pull request by estebanhdzc**
### 🎫 Ticket

[ET-1607] 

### 🗒️ Description

- Added customer name to stripe description payment intent

### 🎥 Artifacts 
https://www.loom.com/share/55a063fd817a4b7abc24d5cafd182eae

### ✔️ Checklist
- [x] I've included a changelog entry in the readme.txt file. 
- [x] My code is tested. 
- [x] My code has [proper inline documentation](https://the-events-calendar.github.io/products-engineering/docs/code-standards/).

[ET-1607]: https://stellarwp.atlassian.net/browse/ET-1607?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/Tickets/Commerce/Gateways/Stripe/Payment_Intent.php [hunk 1]
```diff
@@ -193,8 +193,9 @@ public static function get( $payment_intent_id ) {
 	 * @return array|\WP_Error|null
 	 */
 	public static function update( $payment_intent_id, $data ) {
-		$query_args = [];
-		$args       = [
+		$query_args            = [];
+		$data[ 'description' ] = $data[ 'metadata' ][ 'purchaser_name' ];
```
[the remark this answers is not in the document]
Comment (estebanhdzc) [reply src/Tickets/Commerce/Gateways/Stripe/Payment_Intent.php]: Fixed

[the remark this answers is not in the document]
Comment (estebanhdzc) [reply src/Tickets/Commerce/Gateways/Stripe/Payment_Intent.php]: Fixed

[the remark this answers is not in the document]
Comment (estebanhdzc) [reply src/Tickets/Commerce/Gateways/Stripe/Payment_Intent.php]: Fixed

**Comment by redscar** [comment]
I think you may be better off moving code into the `update_payment_intent` similar to how the `receipt_email` is being added.

https://github.com/the-events-calendar/event-tickets/blob/release/G24.akita/src/Tickets/Commerce/Gateways/Stripe/Payment_Intent_Handler.php#L109

**Review by redscar — APPROVED** [review]

**Review by rafsuntaskin — DISMISSED** [review]
We already have this data in the [metadata](https://share.cleanshot.com/KqDWgqwD) section. Are we sure we want to put this in description as well?

cc @juanfra

**Comment by estebanhdzc** [comment]
@rafsuntaskin I've notice that, but I don't know if the intention of the ticket is to allow help the lazy people to see faster this information, because in the metadata you have to dig it

cc @juanfra

File: src/Tickets/Commerce/Gateways/Stripe/Payment_Intent_Handler.php [hunk 2]
```diff
@@ -118,10 +118,13 @@ public function update_payment_intent( $data, \WP_Post $order ) {
 			if ( is_user_logged_in() ) {
 				$user                  = wp_get_current_user();
 				$body['receipt_email'] = $user->get( 'user_email' );
+				$customer_name         = $user->get( 'first_name' ) ? $user->get( 'first_name' )  . ' ' . $user->get( 'last_name' ) : $user->get( 'display_name' );
```
[the remark this answers is not in the document]
Comment (estebanhdzc) [reply src/Tickets/Commerce/Gateways/Stripe/Payment_Intent_Handler.php]: Fixed

**Comment by estebanhdzc** [comment]
> I think you may be better off moving code into the `update_payment_intent` similar to how the `receipt_email` is being added.
> 
> https://github.com/the-events-calendar/event-tickets/blob/release/G24.akita/src/Tickets/Commerce/Gateways/Stripe/Payment_Intent_Handler.php#L109

@redscar I have applied your suggestion, thanks for the heads up!

**Review by redscar — COMMENTED** [review]
Looking good. The only other thing I would do is update the `@since`.

File: readme.txt [hunk 3]
```diff
@@ -204,7 +204,7 @@ Check out our extensive [knowledgebase](https://evnt.is/18wm) for articles on us
 * Fix - Resolved an issue where Order Status was not populated when exporting the Attendee List using Tickets Commerce. [ET-1883]
 * Tweak - Update default footer text of Tickets Emails to include link to website. [ET-1971]
 * Fix - Update usage of `method_exists()` to comply with PHP 8.1 standards. [ET-1759]
-* Tweak - Customer name appears now as description of a stripe payment. [ET-1607]
+* Tweak - Customer name appears now as description of a stripe payment. Added `tec_tickets_commerce_stripe_update_payment_description` filter. [ET-1607]
```
Comment (redscar) [inline readme.txt:207]:
```suggestion
* Tweak - Customer name appears now as description of a Stripe payment. Added `tec_tickets_commerce_stripe_update_payment_description` filter. [ET-1607]
```

Comment (estebanhdzc) [reply readme.txt]: Thanks @redscar ! I have applied your suggestion

**Review by redscar — APPROVED** [review]

**Review by redscar — APPROVED** [review]

**Comment by estebanhdzc** [comment]
@rafsuntaskin can you take a look please? Github is preventing me from merging until you review it. Thanks

File: src/Tickets/Commerce/Gateways/Stripe/Payment_Intent_Handler.php [hunk 4]
```diff
@@ -252,4 +255,47 @@ static function ( $item ) {
...
+			$user           = wp_get_current_user();
+			$purchaser_name = $user->get( 'first_name' ) ? $user->get( 'first_name' ) . ' ' . $user->get( 'last_name' ) : $user->get( 'display_name' );
+		}
+
+		$tickets_in_order           = implode( ', ', array_unique( array_filter( wp_list_pluck( $order->items, 'ticket_id' ) ) ) );
+		$events_in_order            = array_unique( array_filter( wp_list_pluck( $order->items, 'event_id' ) ) );
+		$post_id                    = current( $events_in_order );
+		$post                       = get_post( $post_id );
+		$post_labels                = get_post_type_labels( get_post_type_object( $post->post_type ) );
+		$payment_intent_description = sprintf( '[%1$s: %2$s] [Ticket: %3$s] [Order: %4$s] %5$s - %6$s - %7$s', $post_labels->singular_name, $post_id, $tickets_in_order, $order->ID, $body['metadata']['event_name'], $body['metadata']['ticket_names'], $purchaser_name );
```
Comment (codingmusician) [inline src/Tickets/Commerce/Gateways/Stripe/Payment_Intent_Handler.php:284]: Let's lay this out a little nicer, pull the correct terminology for "Ticket" and make sure "Order" is translatable:
```suggestion
		$payment_intent_description = sprintf(
			'[%1$s: %2$s] [%3$s: %4$s] [%5$s: %6$s] %7$s - %8$s - %9$s',
			$post_labels->singular_name,
			$post_id,
			tribe_get_ticket_label_singular( 'stripe_payment_intent_description' ),
			$tickets_in_order,
			__( 'Order', 'event-tickets' ),
			$order->ID,
			$body['metadata']['event_name'],
			$body['metadata']['ticket_names'],
			$purchaser_name
		);
```

Comment (estebanhdzc) [reply src/Tickets/Commerce/Gateways/Stripe/Payment_Intent_Handler.php]: Thanks @codingmusician ! I have applied your suggestion.

**Review by rafsuntaskin — APPROVED** [review]

**Review by codingmusician — APPROVED** [review]

**Pull request merged**